src/Tasks/Microsoft.NET.Build.Tasks/Microsoft.NET.Build.Tasks.csproj

```diff
@@ -61,8 +61,21 @@
 
   <!-- Packages that are in-box for .NET Core, so we only need to reference them for .NET Framework -->
   <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
-    <PackageReference Include="System.Reflection.Metadata" Version="$(SystemReflectionMetadataVersion)" />
-    
+    <!--
+        MSBuild depends on a new version of SRM/SCI; HostModel a very old one.
+        In order to keep working with MSBuild 17.2, which had binding redirects
+        for SCI 0-5.0.0.0 but not 6, this task must compile against 5, even
+        though that's not transitively correct. Nothing uses updated API
+        surface so it all works ok.
+
+        To accomplish this, explicitly reference the current version of the
+        packages, but exclude assets so it doesn't get passed to the compiler.
+        Also download an old version and (later in the target ReferenceOlderSCIandSRM)
+        pass them to the compiler. -->
+    <PackageReference Include="System.Reflection.Metadata" Version="$(SystemReflectionMetadataVersion)" IncludeAssets="none" />
+    <PackageReference Include="System.Collections.Immutable" Version="6.0.0" IncludeAssets="none" />
+    <PackageDownload Include="System.Collections.Immutable" Version="[5.0.0]" GeneratePathProperty="true" />
+    <PackageDownload Include="System.Reflection.Metadata" Version="[5.0.0]" GeneratePathProperty="true" />
```

Comment:
Looks like you're not using the properties generated by GeneratePathProperty below. Was that intentional?

Reply:
IIRC it didn't actually produce the properties as I expected. In which case I probably should have removed the attributes here.